app/views/ChooseProvider.js

```diff
@@ -215,33 +199,59 @@ class ChooseProviderScreen extends Component {
     );
   }
 
+  toggleFilterAuthoritesByGPSHistory() {
+    this.filterAuthoritesByGPSHistory({
+      val: !this.state.isAuthorityFilterActive,
+    });
+  }
+
+  async filterAuthoritesByGPSHistory(isAuthorityFilterActive) {
+    await this.fetchAuthoritiesList(isAuthorityFilterActive.val);
+    this.setState({ isAuthorityFilterActive: isAuthorityFilterActive.val });
+  }
+
+  async toggleAutoSubscribe() {
+    this.setState(
+      prevState => ({
+        isAutoSubscribed: !prevState.isAutoSubscribed,
+      }),
+      async () => {
+        this.state.isAutoSubscribed
+          ? await HCAService.enableAutoSubscription()
+          : await HCAService.disableAutoSubscription();
+      },
+    );
+  }
+
   render() {
     return (
       <NavigationBarWrapper
         title={languages.t('label.choose_provider_title')}
         onBackPress={this.backToMain.bind(this)}>
         <View style={styles.main}>
-          <Typography style={styles.headerTitle}>
+          <Typography style={styles.headerTitle} use={Type.Headline2}>
```

Comment:
This is cool but in future just `use="headline2"` is probably fine, in Typescript it would allow it.